src/components/themed/MaybeCustomNodesSetting.js

```diff
@@ -0,0 +1,134 @@
+import { SettingsTappableRow } from './SettingsTappableRow.js'
+
+type CustomNodesSetting = {
+  enableCustomNodes: boolean,
+  customNodes: string[]
+}
+
+type Props = CurrencySettingProps<CustomNodesSetting>
+
+function CustomNodesSettingComponent(props: Props) {
+  const { defaultSetting, setting, onUpdate } = props
+  const { enableCustomNodes, customNodes } = setting
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  async function handleToggleEnabled(): Promise<void> {
+    await onUpdate({
+      enableCustomNodes: !enableCustomNodes,
+      customNodes: customNodes.length > 0 ? customNodes : defaultSetting.customNodes
+    })
+  }
+
+  async function handleDeleteNode(i: number): Promise<void> {
+    const list = [...customNodes]
+    list.splice(i, 1)
+
+    await onUpdate({ enableCustomNodes, customNodes: list })
+  }
+
+  function handleEditNode(i?: number): void {
+    async function handleSubmit(text: string) {
+      const list = [...customNodes]
+      if (i == null) list.push(text)
+      else list[i] = text
+
+      await onUpdate({ enableCustomNodes, customNodes: list })
+      return true
+    }
+
+    Airship.show(bridge => (
+      <TextInputModal
+        autoCorrect={false}
+        bridge={bridge}
+        initialValue={i == null ? '' : customNodes[i]}
+        inputLabel={s.strings.settings_custom_node_url}
+        title={s.strings.settings_edit_custom_node}
+        onSubmit={handleSubmit}
+      />
+    ))
+  }
+
+  return (
+    <>
+      <SettingsHeaderRow label={s.strings.settings_custom_nodes_title} />
+      <SettingsSwitchRow label={s.strings.settings_enable_custom_nodes} value={enableCustomNodes} onPress={handleToggleEnabled} />
+      {!enableCustomNodes ? null : (
+        <>
+          {customNodes.map((server, i) => (
+            <SettingsTappableRow key={`row${i}`} action="delete" onPress={() => handleDeleteNode(i)}>
+              <TouchableOpacity onPress={() => handleEditNode(i)} style={styles.labelContainer}>
+                <Text style={styles.labelText}>{server}</Text>
+              </TouchableOpacity>
+            </SettingsTappableRow>
+          ))}
+          <SettingsTappableRow action="add" label={s.strings.settings_add_custom_node} onPress={handleEditNode} />
+        </>
+      )}
+    </>
+  )
+}
+
+const getStyles = cacheStyles(theme => ({
+  // We use a hack to make the text tappable separately from the switch.
+  labelContainer: {
+    flexGrow: 10,
+    flexShrink: 1,
+    // Stretch outward to cover the row:
+    margin: -theme.rem(0.5),
+    padding: theme.rem(0.5)
+  },
+  labelText: {
+    color: theme.primaryText,
+    flexShrink: 1,
+    fontFamily: theme.fontFaceDefault,
+    fontSize: theme.rem(1),
+    paddingHorizontal: theme.rem(0.5),
+    textAlign: 'left'
+  }
+}))
+
+const asCustomNodesSetting: Cleaner<CustomNodesSetting> = asObject({
+  enableCustomNodes: asOptional(asBoolean, false),
+  customNodes: asArray(asString)
```

Comment:
In `edge-currency-plugins`, the name of the field is `serverList`: https://github.com/EdgeApp/edge-currency-plugins/blob/master/src/common/plugin/pluginState.ts#L23

Reply:
OK, so your plugin needs to be using a cleaner. Also, we talked about changing the name so the upgrade goes smoothly. We don't want the new plugin trying to talk to Electrum servers leftover from the old plugin.

I am open to changing the shape of the cleaner, maybe naming the fields `customServers` and `enableCustomServers` if you prefer, but we want to agree before merging.

Reply:
https://github.com/EdgeApp/edge-currency-plugins/pull/311

Reply:
I concur with the movement to change to your proposal. Very clean and clear semantics! 👍